### PR TITLE
Bug/ms 1789/removing old scrips refing mdm cred

### PR DESCRIPTION
**Pull request by darkbushido**
Removing the following methods
(`grep -roh report_cred . | wc -w`) instances
store_cred (33)
crews
each_cred (25)
find_or_create_cred (8)
report_auth_info
report_auth (41)
report_cred (373)

this should not be landed until the methods are no longer referenced.

### CODE REVIEW

**Comment by bcook-r7** [comment]
What's the plan to remove the remaining method calls?

File: lib/msf/core/auxiliary/report.rb [hunk 1]
```diff
@@ -474,68 +474,71 @@ def store_local(ltype=nil, ctype=nil, data=nil, filename=nil)
...
-      end
-    end
-    print_status "Collecting #{cred_opts[:user]}:#{cred_opts[:pass]}"
-    framework.db.report_auth_info(cred_opts)
-  end
+  # # Takes a credential from a script (shell or meterpreter), and
+  # # sources it correctly to the originating user account or
+  # # session. Note that the passed-in session ID should be the
+  # # Session.local_id, which will be correlated with the Session.id
+  # def store_cred(opts={})
```
Comment (egypt) [inline lib/msf/core/auxiliary/report.rb]: *Please* don't just comment out code. If we're removing it, let's really remove it.

Comment (darkbushido) [inline lib/msf/core/auxiliary/report.rb]: Yes, but store cred is used in 33 other places that werent fixed and will break if this is merged. This should not be merged until the other places have been fixed.
I'm going to leave the commented code up there until the other places its used are cleaned up then the commented code can be removed.

**Comment by egypt** [comment]
I'm not sure where you're getting these numbers. As far as I can tell none of these are referenced except `report_auth_info`. This is what I get:
```
egypt metasploit-framework:upstream-master $ git grep store_cred
lib/msf/core/auxiliary/report.rb:  def store_cred(opts={})
spec/lib/msf/core/auxiliary/cisco_spec.rb:    def store_cred(hsh=nil)
```
```
egypt metasploit-framework:upstream-master $ git grep 'each_cred\b'
lib/msf/core/db_manager/cred.rb:  def each_cred(wspace=workspace,&block)
spec/support/shared/examples/msf/db_manager/cred.rb:  it { is_expected.to respond_to :each_cred }
```

`report_cred` appears to be all false positives because lots of modules use that method name as a wrapper around the `Credential` creation stuff.

**Comment by darkbushido** [comment]
Closing this until its in a land-able state.

**Pull request closed without merging**